### PR TITLE
[Protocol3] reimplement user staking pool

### PR DESCRIPTION
**Pull request by dong77**

### CODE REVIEW

File: packages/loopring_v3/contracts/iface/IProtocolFeeManager.sol [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            uint remainingDev,
+            uint remainingReward
+        );
+
+    /// @dev Set Oedax address, only callable by the owner.
+    /// @param _oedaxAddress The address of Oedax contract.
+    function setOedax(address _oedaxAddress) external;
+
+    /// @dev Permanently disallow owner to withdraw non-LRC protocol fees, only callable by the owner.
+    function disableOwnerWithdrawaldisableOwnerWithdrawal() external;
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/iface/IProtocolFeeManager.sol]: Should be just disableOwnerWithdrawal

File: packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol [hunk 2]
```diff
@@ -183,48 +196,53 @@ contract ProtocolFeeManager is IProtocolFeeManager, Claimable
...
-        //     tokenS,
-        //     auctionAddr
-        // );
+
+        IOedax oedax = IOedax(oedaxAddress);
+        uint ethStake = oedax.creatorEtherStake();
+
+        auctionAddr = oedax.createAuction.value(ethStake)(
+            token,  // askToken
+            tokenB, // bidToken
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol]: What if token == tokenB?

File: packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol [hunk 3]
```diff
@@ -183,48 +196,53 @@ contract ProtocolFeeManager is IProtocolFeeManager, Claimable
...
+            minBidAmount,
+            P,
+            S,
+            M,
+            T,
+            T * 2
+        );
+
+        if (token == address(0)) {
+            auctionAddr.transfer(amount);
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol]: `address.transfer` has a gas limit of 2300 gas which is probably too low to do the additional logic in the default function. Transferring the ETH like this doesn't have that gas limit:
```
(success, ) = auctionAddr.call.value(amount)("");
```

File: packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol (hunk 2 above)
Comment (dong77) [reply packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol]: when token == tokenB then the oedax auction will fail. This is checked inside the oedax contract.

Comment (dong77) [reply packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol]: I added another check.

File: packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol (hunk 3 above)
Comment (dong77) [reply packages/loopring_v3/contracts/impl/ProtocolFeeManager.sol]: great.

**Pull request merged**